### PR TITLE
docs: Clarify potential gotcha when overriding serialize in rtdb

### PR DESCRIPTION
**Pull request by julienr**
TLDR: The rtdb  `childRemoved` handler uses `indexForKey` to find the element to delete. This function assumes that a `.key` property is defined on objects, which won't be the case if you override the `serialize` option. This can lead to bugs when removing from array.

For example, I had the following, which doesn't define `.key`:

```
      await bindFirebaseRef('events', db.ref(FB_EVENT_PREFIX + aoiName + '/'), {
        serialize: (snapshot: firebase.database.DataSnapshot) => {
          const val = snapshot.val()
          val.date = new Date(val.date)
          return val
        }
      })
```

With this code, whenever I deleted an element from my array, it was always the last one that got removed. Took me a bit of time to figure out why deletion wasn't working, hence this documentation improvement.

The reason is that if `.key` is not defined, `indexForKey` [will return -1](https://github.com/vuejs/vuefire/blob/c781f6a98ef9216246ef4b8e8ff7cf5f4675e24c/packages/%40posva/vuefire-core/src/rtdb/utils.ts#L38) which ultimately will gets passed to [Array.slice](https://github.com/vuejs/vuefire/blob/c781f6a98ef9216246ef4b8e8ff7cf5f4675e24c/packages/vuefire/src/rtdb.ts#L27) which ends up removing the last item.

A better fix might be to throw an error instead of returning -1 in `indexForKey`. Happy to work on this if that's deemed better.

### CODE REVIEW

**Comment by posva** [comment]
Thanks for this, I added a similar comment to the new docs
